internal/incident/incidents.go

```diff
@@ -42,10 +42,13 @@ func GetCurrent(
 
 		err := db.QueryRowx(db.Rebind(db.BuildSelectStmt(ir, ir)+` WHERE "object_id" = ? AND "recovered_at" IS NULL`), obj.ID).StructScan(ir)
 		if err != nil && err != sql.ErrNoRows {
-			return nil, false, fmt.Errorf("incident query failed with: %w", err)
+			logger.Errorw("Failed to load incident from database", zap.String("object", obj.DisplayName()), zap.Error(err))
+
+			return nil, false, errors.New("failed to load incident from database")
 		} else if err == nil {
 			incident.incidentRowID = ir.ID
 			incident.StartedAt = ir.StartedAt.Time()
+			incident.logger = logger.With(zap.String("object", obj.DisplayName()), zap.String("incident", incident.String()))
```

Comment:
You're initializing `incident.logger` only here but not when creating a new incident. In that case, it will panic.